features/storage/FEATURE_STORAGE/cfstore/configuration-store/configuration_store.h

```diff
@@ -163,7 +163,7 @@ typedef struct _ARM_CFSTORE_STATUS {
     ARM_CFSTORE_HANDLE (__name) = (ARM_CFSTORE_HANDLE) (__name##_buf_cFsToRe);  \
     memset((__name##_buf_cFsToRe), 0, CFSTORE_HANDLE_BUFSIZE)
 
-#if defined __MBED__ && (defined TOOLCHAIN_GCC_ARM || defined TOOLCHAIN_ARMC6)
+#if defined __MBED__ && (defined TOOLCHAIN_GCC_ARM || (defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)))
```

Comment:
> __ARMCC_VERSION >= 6010050

Seems weird to have this.

Reply:
@cmonr - Thanks for pointing that out. Since I was replacing "defined TOOLCHAIN_ARMC6" I wanted to make sure the bounded code gets applied to ARMC6 versions only. Thats why we have that extra version check.